### PR TITLE
Add 'Pause On Overlay' feature

### PR DESCRIPTION
**Pull request by therealpaulgg**
Adds feature for pause on overlay.

- Current implementation adds a separate switch for 'Pause on overlay'
- Only works when steam button is pressed and focus is lost.
- Does not pause on bringing up quick menu or power menu.
- Games are only paused when there is a combination of focus loss and steam button is pressed.

image

### CODE REVIEW

File: src/backend.ts [hunk 1]
```diff
@@ -311,14 +321,52 @@ export function setupSuspendResumeHandler(): () => void {
 export function setupFocusChangeHandler(): () => void {
   let appIsStartingUp: boolean = false;
   let lastPid = 0;
+  let lastAppid = 0;
+  let validKeyEvent: SystemKeyEvent | null = null;
+  SteamClient.System.UI.RegisterForSystemKeyEvents((e) => {
```
Comment (popsUlfr) [inline src/backend.ts:326]: It's unfortunate that this doesn't seem to return an unregister function. Maybe to alleviate that a bit, the inner logic could be wrapped in another function that can be set to the noop (`() => {}`) on dismount.

File: src/backend.ts [hunk 2]
```diff
@@ -311,14 +321,52 @@ export function setupSuspendResumeHandler(): () => void {
...
+    // Have not found any race condition issues with this approach since the key event fires long before focus change
+    validKeyEvent = e;
```
Comment (popsUlfr) [inline src/backend.ts:328]: I feel like to be on the safe side, `validKeyEvent` should be forcibly set to null after a delay where the STEAM button has not been pressed. I think the lodash `debounce` function could do the job.
Something like:
```ts
const debouncedValidKeyEventReset = debounce(() => { validKeyEvent = null; }, 1000);
...
if (e.eKey === 0) {
  validKeyEvent = e;
  debouncedValidKeyEventReset();
}
```

File: src/backend.ts [hunk 3]
```diff
@@ -311,14 +321,52 @@ export function setupSuspendResumeHandler(): () => void {
...
+          fce.focusedApp.pid === lastPid &&
+          fce.focusedApp.appid === lastAppid
+        )
+          return;
         lastPid = fce.focusedApp.pid;
+        lastAppid = fce.focusedApp.appid;
+        // Only pause if the user has enabled it
+        if (
+          fce.focusedApp.appid === 769 &&
+          (await loadSettings()).overlayPause &&
```
Comment (popsUlfr) [inline src/backend.ts:347]: I think it would be preferable to have the value dependent on the promise as last test. It doesn't make a lot of difference in the grand scheme of things but I would put simple tests (number and string) before more complex ones so that there's an opportunity to escape the condition faster :)

File: src/backend.ts [hunk 4]
```diff
@@ -311,14 +321,52 @@ export function setupSuspendResumeHandler(): () => void {
...
+            if (appid) {
+              const appMD = await getAppMetaData(Number(appid));
+              appMD.is_paused = await is_paused(appMD.instanceid);
+              if (appMD.sticky_state) {
+                return;
+              }
+              if (!appMD.is_paused) {
+                appMD.is_paused = await pause(appMD.instanceid);
+              }
+              return;
```
Comment (popsUlfr) [inline src/backend.ts:363]: Are these returns intentional ? Because they kinda shortcircuit the "Pause on focus loss" feature when multiple apps are running.

Ah I guess the problem would be that the app might be unpaused since it is the main one.

File: src/index.tsx [hunk 5]
```diff
@@ -141,6 +143,21 @@ const Content: VFC<{ serverAPI: ServerAPI }> = ({}) => {
...
+          checked={overlayPause}
+          label="Pause on overlay"
+          tooltip="Pause apps when interacting with Steam Overlay."
+          icon={<FaHome />}
+          onChange={async (state) => {
+            const settings = await backend.loadSettings();
+            settings.overlayPause = state;
+            await backend.saveSettings(settings);
+            setOverlayPause(state);
+            backend.resetStickyPauseStates();
```
Comment (popsUlfr) [inline src/index.tsx:157]: Does it make sense to reset the sticky states when enabling/disabling this feature ? The sticky states are enabled when "Pause on focus loss" is on but it might be off and "Pause on overlay" on.

File: src/backend.ts [hunk 6]
```diff
@@ -311,14 +321,52 @@ export function setupSuspendResumeHandler(): () => void {
...
+            validKeyEvent = null;
+            const appid = await appid_from_pid(fce.focusedApp.pid);
+            if (appid) {
+              const appMD = await getAppMetaData(Number(appid));
+              appMD.is_paused = await is_paused(appMD.instanceid);
+              if (appMD.sticky_state) {
+                return;
+              }
+              if (!appMD.is_paused) {
+                appMD.is_paused = await pause(appMD.instanceid);
```
Comment (popsUlfr) [inline src/backend.ts:361]: At this point it would make sense to pause all running apps no ? There should be a tighter cohesion between "Pause on focus loss" and "Pause on overlay".

**Review by popsUlfr — CHANGES_REQUESTED** [review]
I think there should be a tighter integration with the "Pause on focus loss" otherwise they break each other.

There's an argument here in favour of having the "Pause on overlay" as a sub option of "Pause on focus loss" BUT it also makes sense as a separate feature.
The feature should most likely put all processes to sleep when the overlay is up and thus follow the same logic as the "Pause on focus loss" feature with specific conditions.

For the sticky states, I guess it makes sense that "Pause on overlay" would reset them when enabled/disabled if the user is allowed to override the state in this mode.

Unless "Pause on focus loss" is also enabled, invoking the overlay should pause all apps and when dismissing the overlay they should all be resumed unless the sticky states state otherwise ?

About the sticky states, I think they should only be reset if both "Pause on focus loss" and "Pause on overlay" are disabled.

File: src/backend.ts (hunk 1 above)
Comment (therealpaulgg) [reply src/backend.ts]: Added:

```ts
let keyEventFunction = (e: SystemKeyEvent) => {
    // Have not found any race condition issues with this approach since the key event fires long before focus change
    validKeyEvent = e;
  }
  const unregisterSystemKeyEvents = () => {
    keyEventFunction = () => {};
  }
  SteamClient.System.UI.RegisterForSystemKeyEvents((e) => keyEventFunction(e));
```

File: src/backend.ts (hunk 3 above)
Comment (therealpaulgg) [reply src/backend.ts]: Added!

File: src/backend.ts (hunk 2 above)
Comment (therealpaulgg) [reply src/backend.ts]: Added here:

```ts
let keyEventFunction = (e: SystemKeyEvent) => {
    const cancelDebouncedEvent = debounce(() => {
      validKeyEvent = null;
    }, 1000);
    if (e.eKey === 0) {
      // Have not found any race condition issues with this approach since the key event fires long before focus change
      validKeyEvent = e;
      cancelDebouncedEvent();
    }
  };
```

File: src/backend.ts (hunk 4 above)
Comment (therealpaulgg) [reply src/backend.ts:373]: I ran into an issue where apps would not unpause themselves if I didn't return sooner, I believe that's why I added the short circuit

File: src/index.tsx (hunk 5 above)
Comment (therealpaulgg) [reply src/index.tsx:157]: I'm ok with not resetting sticky state.

**Comment by therealpaulgg** [comment]
The more I think about it, I prefer thinking of the 'pause on overlay' as an extension of the pause on focus feature and they should be used together. While they technically could be used separately, it would be a little weird in my opinion to have a case where you have all your apps running in the background AND you pause them all on overlay. Making it a sub-option makes more sense to me currently.

Basically when the overlay is open, all apps (aside from stickied ones) should be paused. When it's closed, the only app that should be resumed is the one currently in focus. 

I'll make some tweaks and do a bit more testing and see how this works.

File: src/backend.ts (hunk 6 above)
Comment (therealpaulgg) [reply src/backend.ts:371]: I think technically it doesn't make a difference because all apps *should* be paused anyway. But to be safe I agree, I'll pause all running apps.

File: src/backend.ts (hunk 4 above)
Comment (therealpaulgg) [reply src/backend.ts:373]: Also the later logic would have the app unpause immediately after the pause is triggered, yes.

**Comment by therealpaulgg** [comment]
Ok, here's where we're at:
image

Made the overlay fully dependent on the focus loss feature, and if focus loss gets turned off so does overlay.

I am having trouble figuring out styling here. There's an `indentLevel` prop on ToggleField but it seems to do nothing. using a div seems to add padding around the entire element (when I just asked for `padding-left`). 

Any tips?

**Comment by popsUlfr** [comment]
> The more I think about it, I prefer thinking of the 'pause on overlay' as an extension of the pause on focus feature and they should be used together. While they technically could be used separately, it would be a little weird in my opinion to have a case where you have all your apps running in the background AND you pause them all on overlay. Making it a sub-option makes more sense to me currently.
> 
> Basically when the overlay is open, all apps (aside from stickied ones) should be paused. When it's closed, the only app that should be resumed is the one currently in focus.

I also have a hard time to find real world useful uses for it as a standalone feature. It's something you would expect as being part of the "Pause on focus loss" feature but it could be annoying in some cases to have it on by default for users (for the reasons a talked about a bit earlier).
As an additional option to the "Pause on focus loss" for a more complete pause for everything seems to be the sensible way to go indeed.

> I am having trouble figuring out styling here. There's an indentLevel prop on ToggleField but it seems to do nothing. using a div seems to add padding around the entire element (when I just asked for padding-left).

CSS Loader is using an indentation with arrow for sub-options which looks neat:

Maybe we could follow that too (https://github.com/suchmememanyskill/SDH-CssLoader)

The sub-options in CSS Loader only appear when the root option is on.
For the text maybe it could be something like `Also on overlay` with a short help text about it in the help panel when no app is running ?

**Comment by therealpaulgg** [comment]
I figured it would be something sort of like what they did, thanks for the pointer.

Here is how it looks now:

image

The helper text at the bottom does go a bit off the screen now unfortunately but it's possible to scroll down at least with the touch screen. I personally think it's ok but let me know.

File: src/backend.ts [hunk 7]
```diff
@@ -311,14 +321,69 @@ export function setupSuspendResumeHandler(): () => void {
...
-        if (fce.focusedApp.pid === lastPid) return;
+        if (
+          fce.focusedApp.pid === lastPid &&
+          fce.focusedApp.appid === lastAppid
+        )
+          return;
         lastPid = fce.focusedApp.pid;
+        lastAppid = fce.focusedApp.appid;
+        // Only pause if the user has enabled it
+        if (
```
Comment (popsUlfr) [inline src/backend.ts:355]: This whole logic should be merged with the Pause on focus loss feature, since the feature depends on `autoPause` being on. So after the `if (!(await loadSettings()).autoPause) return;`. This way the ToggleField also doesn't need to be aware of `overlayPause`.

So I think in order to merge all this functionality together we could do it like this (but please correct me if I missed something):

```diff
diff --git a/src/backend.ts b/src/backend.ts
index 662ee26..c25ce39 100644
--- a/src/backend.ts
+++ b/src/backend.ts
@@ -351,41 +351,14 @@ export function setupFocusChangeHandler(): () => void {
           return;
         lastPid = fce.focusedApp.pid;
         lastAppid = fce.focusedApp.appid;
-        // Only pause if the user has enabled it
-        if (
-          fce.focusedApp.appid === 769 &&
-          fce.focusedApp.strExeName !== "steamwebhelper" &&
-          (await loadSettings()).overlayPause
-        ) {
-          // Key event must be if the user pressed the 'STEAM' button. Don't pause for other buttons
-          if (validKeyEvent?.eKey === 0) {
-            validKeyEvent = null;
-            await Promise.all(
-              (Router.RunningApps as AppOverviewExt[]).map(async (a) => {
-                const appMD = await getAppMetaData(Number(a.appid));
-                // if the sticky pause state is on for this app don't do anything to it
-                if (appMD.sticky_state) {
-                  return a;
-                }
-                appMD.is_paused = await is_paused(appMD.instanceid);
-                // the app is not in focus
-                if (!appMD.is_paused) {
-                  appMD.is_paused = await pause(appMD.instanceid);
-                  appMD.pause_state_callbacks.forEach((cb) =>
-                    cb(appMD.is_paused)
-                  );
-                }
-                return a;
-              })
-            );
-            return;
-          } else {
-            return;
-          }
-        }
-
         if (!(await loadSettings()).autoPause) return;
         // AppID 769 is the Steam Overlay or a non-steam app
+        // Key event must be if the user pressed the 'STEAM' button. Don't pause for other buttons
+        const overlayPause =
+          fce.focusedApp.appid === 769 &&
+          fce.focusedApp.strExeName !== "steamwebhelper" &&
+          validKeyEvent?.eKey === 0 &&
+          (await loadSettings()).overlayPause;
         if (!fce.focusedApp.appid || fce.focusedApp.appid === 769) {
           const appid = await appid_from_pid(fce.focusedApp.pid);
           if (appid) {
@@ -409,7 +382,7 @@ export function setupFocusChangeHandler(): () => void {
               return a;
             }
             appMD.is_paused = await is_paused(appMD.instanceid);
-            if (appMD.instanceid === fce.focusedApp.pid) {
+            if (!overlayPause && appMD.instanceid === fce.focusedApp.pid) {
               // this is the focused app
               if (appMD.is_paused) {
                 appMD.is_paused = !(await resume(appMD.instanceid));
@@ -418,7 +391,7 @@ export function setupFocusChangeHandler(): () => void {
                 );
               }
             } else {
-              // the app is not in focus
+              // the app is not in focus or the overlay is on
               if (!appMD.is_paused) {
                 appMD.is_paused = await pause(appMD.instanceid);
                 appMD.pause_state_callbacks.forEach((cb) =>

```

The whole `RegisterForFocusChangeEvents` section would look like this:

```ts
  const { unregister: unregisterFocusChangeEvents } =
    SteamClient.System.UI.RegisterForFocusChangeEvents(
      throttle(async (fce: FocusChangeEvent) => {
        // don't try anything while an application is launching or it could pause it midlaunch
        if (appIsStartingUp) return;
        // skip if we already got such an event before
        if (
          fce.focusedApp.pid === lastPid &&
          fce.focusedApp.appid === lastAppid
        )
          return;
        lastPid = fce.focusedApp.pid;
        lastAppid = fce.focusedApp.appid;
        if (!(await loadSettings()).autoPause) return;
        // AppID 769 is the Steam Overlay or a non-steam app
        // Key event must be if the user pressed the 'STEAM' button. Don't pause for other buttons
        const overlayPause =
          fce.focusedApp.appid === 769 &&
          fce.focusedApp.strExeName !== "steamwebhelper" &&
          validKeyEvent?.eKey === 0 &&
          (await loadSettings()).overlayPause;
        if (!fce.focusedApp.appid || fce.focusedApp.appid === 769) {
          const appid = await appid_from_pid(fce.focusedApp.pid);
          if (appid) {
            fce.focusedApp.appid = appid;
          } else {
            return;
          }
        }
        // fce.focusedApp.pid is not the pid of the reaper but of the first child
        const pid = await pid_from_appid(fce.focusedApp.appid);
        if (pid) {
          fce.focusedApp.pid = pid;
        } else {
          return;
        }
        await Promise.all(
          (Router.RunningApps as AppOverviewExt[]).map(async (a) => {
            const appMD = await getAppMetaData(Number(a.appid));
            // if the sticky pause state is on for this app don't do anything to it
            if (appMD.sticky_state) {
              return a;
            }
            appMD.is_paused = await is_paused(appMD.instanceid);
            if (!overlayPause && appMD.instanceid === fce.focusedApp.pid) {
              // this is the focused app
              if (appMD.is_paused) {
                appMD.is_paused = !(await resume(appMD.instanceid));
                appMD.pause_state_callbacks.forEach((cb) =>
                  cb(appMD.is_paused)
                );
              }
            } else {
              // the app is not in focus or the overlay is on
              if (!appMD.is_paused) {
                appMD.is_paused = await pause(appMD.instanceid);
                appMD.pause_state_callbacks.forEach((cb) =>
                  cb(appMD.is_paused)
                );
              }
            }
            return a;
          })
        );
      }, 500)
    );
```

File: src/index.tsx [hunk 8]
```diff
@@ -128,19 +130,40 @@ const Content: VFC<{ serverAPI: ServerAPI }> = ({}) => {
...
           checked={autoPause}
           label="Pause on focus loss"
           tooltip="Pauses apps not in focus when switching between them."
           icon={<FaStream />}
           onChange={async (state) => {
             const settings = await backend.loadSettings();
             settings.autoPause = state;
-            await backend.saveSettings(settings);
             setAutoPause(state);
+            if (!state) {
```
Comment (popsUlfr) [inline src/index.tsx:142]: By changing the logic of the `Pause on overlay` feature in the backend you wouldn't need this anymore. As a side effect it would remember the `Also on overlay` state if `Pause on focus loss` was disabled and re-enabled.

File: src/index.tsx [hunk 9]
```diff
@@ -164,6 +187,13 @@ const Content: VFC<{ serverAPI: ServerAPI }> = ({}) => {
           <FaPlay color="deepskyblue" />, <FaPause color="deepskyblue" />. To
           reset, disable and re-enable <em>Pause on focus loss</em>.
           <br />
+          <strong>
+            <em>- Also on overlay</em>
+          </strong>
+          <br />
+          Pauses apps when interacting with Steam Overlay. Requires pause on
```
Comment (popsUlfr) [inline src/index.tsx:194]: `Requires pause on focus loss to be enabled.` Could potentially be removed since it will be functionally apparent when interacting with the options.

Maybe a slightly smaller font could be achieved with `font-size` styling (percent or em) so everything fits or a down arrow could be fixed with absolute positioning to make it clear to the user that there's is more text ?

**Review by popsUlfr — CHANGES_REQUESTED** [review]
Very nice!

The last piece of the puzzle would be merging of the functionality with "Pause on focus loss" as I commented it on the code.

For the help text overflowing maybe a slight font size modification or putting a down arrow maybe could make it clear to the user that there is more text ?

File: src/backend.ts [hunk 10]
```diff
@@ -311,14 +321,69 @@ export function setupSuspendResumeHandler(): () => void {
...
+          fce.focusedApp.appid === 769 &&
+          fce.focusedApp.strExeName !== "steamwebhelper" &&
```
Comment (popsUlfr) [inline src/backend.ts:357]: I wonder if the test for not "steamwebhelper" is really desirable here ? For instance, when I have an app running but I go to my Home screen `strExeName` will be "steamwebhelper" but I would still want the "Pause on overlay" feature to apply. Just checking if `fce.focusedApp.appid` is `769` might be enough.

File: src/backend.ts [hunk 11]
```diff
@@ -311,14 +321,69 @@ export function setupSuspendResumeHandler(): () => void {
 export function setupFocusChangeHandler(): () => void {
   let appIsStartingUp: boolean = false;
   let lastPid = 0;
+  let lastAppid = 0;
+  let validKeyEvent: SystemKeyEvent | null = null;
+  let keyEventFunction = (e: SystemKeyEvent) => {
+    const cancelDebouncedEvent = debounce(() => {
+      validKeyEvent = null;
+    }, 1000);
+    if (e.eKey === 0) {
```
Comment (popsUlfr) [inline src/backend.ts:330]: Ah, additionally `validKeyEvent` could be set to `null` in the `else` to be extra safe.

File: src/index.tsx (hunk 9 above)
Comment (popsUlfr) [reply src/index.tsx:194]: `style={{ fontSize: "80%" }}` on the main `<div>` of the help text looks good.

File: src/backend.ts (hunk 10 above)
Comment (popsUlfr) [reply src/backend.ts:357]: Tested a bit more and it all seems to work well with just `fce.focusedApp.appid === 769`, no `fce.focusedApp.strExeName !== "steamwebhelper"`.

File: src/backend.ts (hunk 11 above)
Comment (popsUlfr) [reply src/backend.ts:330]:
```ts
    if (e.eKey === 0) {
      // Have not found any race condition issues with this approach since the key event fires long before focus change
      validKeyEvent = e;
      cancelDebouncedEvent();
    } else {
      cancelDebouncedEvent.cancel();
      validKeyEvent = null;
    }
```

File: src/index.tsx (hunk 9 above)
Comment (popsUlfr) [reply src/index.tsx:194]:
```ts
        <div style={{ fontSize: "80%" }}>
          <strong>
            <em>- Pause before Suspend</em>
          </strong>
          <br />
          Pauses all apps before system suspend. May fix audio issues.
          <br />
          <strong>
            <em>- Pause on focus loss</em>
          </strong>
          <br />
          Pauses apps not in focus automatically when switching between them.
          Changing the state of an app in this mode will sticky them{" "}
          <FaPlay color="deepskyblue" />, <FaPause color="deepskyblue" />. To
          reset, disable and re-enable <em>Pause on focus loss</em>.
          <br />
          <strong>
            <em>- Also on overlay</em>
          </strong>
          <br />
          Pauses apps when interacting with Steam Overlay. Requires pause on
          focus loss to be enabled.
          <br />
          <strong>
            <em>Applications will appear here.</em>
          </strong>
        </div>
```

**Comment by popsUlfr** [comment]
So the complete changes on top of the current pull request I'm testing right now look like this:

```diff
diff --git a/src/backend.ts b/src/backend.ts
index 662ee26..e1df29d 100644
--- a/src/backend.ts
+++ b/src/backend.ts
@@ -331,6 +331,9 @@ export function setupFocusChangeHandler(): () => void {
       // Have not found any race condition issues with this approach since the key event fires long before focus change
       validKeyEvent = e;
       cancelDebouncedEvent();
+    } else {
+      cancelDebouncedEvent.cancel();
+      validKeyEvent = null;
     }
   };
   const unregisterSystemKeyEvents = () => {
@@ -351,41 +354,13 @@ export function setupFocusChangeHandler(): () => void {
           return;
         lastPid = fce.focusedApp.pid;
         lastAppid = fce.focusedApp.appid;
-        // Only pause if the user has enabled it
-        if (
-          fce.focusedApp.appid === 769 &&
-          fce.focusedApp.strExeName !== "steamwebhelper" &&
-          (await loadSettings()).overlayPause
-        ) {
-          // Key event must be if the user pressed the 'STEAM' button. Don't pause for other buttons
-          if (validKeyEvent?.eKey === 0) {
-            validKeyEvent = null;
-            await Promise.all(
-              (Router.RunningApps as AppOverviewExt[]).map(async (a) => {
-                const appMD = await getAppMetaData(Number(a.appid));
-                // if the sticky pause state is on for this app don't do anything to it
-                if (appMD.sticky_state) {
-                  return a;
-                }
-                appMD.is_paused = await is_paused(appMD.instanceid);
-                // the app is not in focus
-                if (!appMD.is_paused) {
-                  appMD.is_paused = await pause(appMD.instanceid);
-                  appMD.pause_state_callbacks.forEach((cb) =>
-                    cb(appMD.is_paused)
-                  );
-                }
-                return a;
-              })
-            );
-            return;
-          } else {
-            return;
-          }
-        }
-
         if (!(await loadSettings()).autoPause) return;
         // AppID 769 is the Steam Overlay or a non-steam app
+        // Key event must be if the user pressed the 'STEAM' button. Don't pause for other buttons
+        const overlayPause =
+          fce.focusedApp.appid === 769 &&
+          validKeyEvent?.eKey === 0 &&
+          (await loadSettings()).overlayPause;
         if (!fce.focusedApp.appid || fce.focusedApp.appid === 769) {
           const appid = await appid_from_pid(fce.focusedApp.pid);
           if (appid) {
@@ -409,7 +384,7 @@ export function setupFocusChangeHandler(): () => void {
               return a;
             }
             appMD.is_paused = await is_paused(appMD.instanceid);
-            if (appMD.instanceid === fce.focusedApp.pid) {
+            if (!overlayPause && appMD.instanceid === fce.focusedApp.pid) {
               // this is the focused app
               if (appMD.is_paused) {
                 appMD.is_paused = !(await resume(appMD.instanceid));
@@ -418,7 +393,7 @@ export function setupFocusChangeHandler(): () => void {
                 );
               }
             } else {
-              // the app is not in focus
+              // the app is not in focus or the overlay is on
               if (!appMD.is_paused) {
                 appMD.is_paused = await pause(appMD.instanceid);
                 appMD.pause_state_callbacks.forEach((cb) =>
diff --git a/src/index.tsx b/src/index.tsx
index aa4d9b0..e32cfe2 100755
--- a/src/index.tsx
+++ b/src/index.tsx
@@ -138,12 +138,8 @@ const Content: VFC<{ serverAPI: ServerAPI }> = ({}) => {
           onChange={async (state) => {
             const settings = await backend.loadSettings();
             settings.autoPause = state;
-            setAutoPause(state);
-            if (!state) {
-              settings.overlayPause = false;
-              setOverlayPause(false);
-            }
             await backend.saveSettings(settings);
+            setAutoPause(state);
             backend.resetStickyPauseStates();
           }}
         />
@@ -171,7 +167,7 @@ const Content: VFC<{ serverAPI: ServerAPI }> = ({}) => {
           </PanelSectionRow>
         ))
       ) : (
-        <div>
+        <div style={{ fontSize: "80%" }}>
           <strong>
             <em>- Pause before Suspend</em>
           </strong>

```

**Comment by popsUlfr** [comment]
It works well. One thing to keep in mind is that using the "Exit game" feature from the overlay will most likely end up force killing the application instead of sending SIGTERM which would have done proper saving and cleanup for the application if it was designed to do that.

**Comment by popsUlfr** [comment]
@therealpaulgg Could you squash your commits together into one and force push if it all looks good to you ? Thanks.

**Comment by therealpaulgg** [comment]
Ok I squashed the commits. I'll be able to test a bit later today if needed.

Noted on steam beta, I'm on normal release so hope it should be ok?

**Comment by popsUlfr** [comment]
> Ok I squashed the commits. I'll be able to test a bit later today if needed.
> 
> Noted on steam beta, I'm on normal release so hope it should be ok?

Thank you so much! I've also updated the dependencies and applied the upstream changes. Also using the `Marquee` component for game titles that are too long to fit to scroll with an animation.

Steam Client Stable still works with decky but Steam Client Beta has changed a bunch of stuff that have killed decky for now.
